backend/open_webui/routers/metrics.py

```
@@ -0,0 +1,477 @@
+
+
+############################
+# Date Range Metrics
+############################
+
+
+@router.get("/range/users")
+async def get_range_metrics(
+    start_date: str,
+    end_date: str,
+    domain: str = None,
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)